src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts

```diff
@@ -29,24 +43,124 @@ export class StickyScrollWidget extends Disposable implements IOverlayWidget {
 	private _lineHeight: number;
 	private _lineNumbers: number[];
 	private _lastLineRelativePosition: number;
+	private _hoverOnLine: number;
+	private _hoverOnColumn: number;
+	private _stickyRangeProjectedOnEditor: IRange;
 
-	constructor(private readonly editor: ICodeEditor) {
+	constructor(
+		private readonly _editor: ICodeEditor,
+		@ILanguageFeaturesService private readonly _languageFeatureService: ILanguageFeaturesService,
+		@IInstantiationService private readonly _instaService: IInstantiationService
+	) {
 		super();
-		this._layoutInfo = this.editor.getLayoutInfo();
+		this._layoutInfo = this._editor.getLayoutInfo();
 		this._rootDomNode = document.createElement('div');
 		this._rootDomNode.className = 'sticky-widget';
 		this._rootDomNode.style.width = `${this._layoutInfo.width - this._layoutInfo.minimap.minimapCanvasOuterWidth - this._layoutInfo.verticalScrollbarWidth}px`;
-
 		this._lineNumbers = [];
 		this._lastLineRelativePosition = 0;
-
-		this._lineHeight = this.editor.getOption(EditorOption.lineHeight);
-		this._register(this.editor.onDidChangeConfiguration(e => {
+		this._hoverOnLine = -1;
+		this._hoverOnColumn = -1;
+		this._stickyRangeProjectedOnEditor = {} as IRange;
```

Comment:
`{} as IRange` doesn't create a valid range it forces a type cast. Better use `undefined` because no instance exists yet